### DISCUSSION TITLE
Creating script to copy tag (or first value) to another tag

### DISCUSSION THREAD

**Discussion by Enn83** [luminescence-software/scripts#13] [Q&A] [answered]
Hello!

I often use your (very useful!) Metatogger software to manage my music library, and this time I want to fill the `"Composer"` tag of all my music, for example to group all music made by the same "Composer" (here `Bob`) but where the artist can be `"Bob"`, `"Bob feat. XYZ"`, etc..

I tried to copy the first tag value from the `"Artist"` tag to the `"Composer"` one, but the tag stay empty. (Below my little script)
I tried replacing the line by `file.SetTag("Composer", "Test");` and here the Composer tag of the selected/active file successfully set to `"Test"`.

Do I made something wrong?

Thanks in advance!

```cs
// Describe here what the script does
// Copy the first value of a tag to another one

using System.Linq;
using Metatogger.Data;

// For each active file ...
foreach (var file in files)

	// ... copy tag value to another tag -- (Do not work ...)
	file.SetTag("Composer", file.GetFirstValue("Album"));

	// ... set "Test" value to another tag -- (Work well)
	//file.SetTag("Composer", "Test");


// The "files" variable contains the collection of audio files checked in Metatogger
// Dictionary<string, List<string>> AudioFile.GetAllTags() => returns all tags of an audio file
// List<string> AudioFile.GetValues(string tag) => returns all values of a tag
// string AudioFile.GetFirstValue(string tag) => returns first tag value for a tag
// AudioFile.SetTag(string tag, string value, bool overwrite = true, bool addIfExists = false) => add, overwrite or remove tag values (if value = null)
// AudioFile.SetTagValue(string tag, string oldValue, string newValue) => replaces the value of a tag by a new value
```

**Comment by CyberSinh** [comment, marked as answer]
By convention, tag name should be in uppercase. 

So, replace:
```cs
file.SetTag("Composer", file.GetFirstValue("Album"));
```
by: 
```cs
file.SetTag("COMPOSER", file.GetFirstValue("ALBUM"));
```
or you can use the `TagName` static class for known tags:
```cs
file.SetTag("COMPOSER", file.GetFirstValue(TagName.Album));
```

But if you're simply copying the value of one tag to another, there's no need to use a script.
Instead, use the manual tag editing tool, and enter the identifier `|ARTIST|` (identifier is case insensitive) as the target value for the COMPOSER tag.

Further information: https://www.luminescence-software.org/en/metatogger/documentation#how-to-add-or-modify-tags

**Reply by Enn83** [reply]
I didn't notice I can also use identifiers like `|ARTIST|` in **[Tag Edit]** tab!
Thanks a lot, it works perfectly 😊! And happy holidays 🎉 !
